src/libcore/iter/mod.rs

```diff
@@ -1643,20 +1708,98 @@ impl<I: DoubleEndedIterator, U, F> DoubleEndedIterator for FlatMap<I, U, F> wher
 ///
 /// [`fuse()`]: trait.Iterator.html#method.fuse
 /// [`Iterator`]: trait.Iterator.html
-#[derive(Clone, Debug)]
 #[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
 #[stable(feature = "rust1", since = "1.0.0")]
-pub struct Fuse<I> {
-    iter: I,
-    done: bool
+pub struct Fuse<I: Iterator> {
```

Comment:
I think this is going to be a breaking change, right? Requiring `I: Iterator` as part of the struct declaration?